Static site generators (#797)
Codecademy/docs: content/general/concepts/static-site-generators/static-site-generators.md

---
Title: 'Static Site Generators'
Description: 'A static site generator (SSG) is a tool used to render web pages on a server at build time, prior to when a client makes a request.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Static Site'

>   - 'Static Site Generator'

Reviewer: ```suggestion
```

However, as the web evolved, websites were no longer limited to providing static content. **[Server-side rendering](https://www.codecademy.com/resources/docs/general/server-side-rendering) (SSR)** was adopted and allowed for content to be “customized” dynamically. **Content Management Systems (CMSs)**, such as WordPress and Drupal, employ an SSR approach and were introduced in response to the growing amount of content that needed to be maintained.

## How SSGs differ from CMSs

Static site generation is a concept that leverages the benefits of static pages and CMSs. Like CMSs, the content (data) and the structure (templates) are decoupled, but instead of the server generating the page at the time of a request, the page is pre-rendered at build time. The page can now be served as a static file in response to a client request, eliminating the time associated with generating the page on demand.

## Examples of SSGs

> Some commonly used SSGs include Jekyll, Hugo, Gatsby, and [Next.js](https://www.codecademy.com/resources/docs/general/next-js). Visit [staticgen.com](http://staticgen.com) for a comprehensive list of SSGs available in different languages.

Reviewer: ```suggestion
Some commonly used SSGs include Jekyll, Hugo, Gatsby, and [Next.js](https://www.codecademy.com/resources/docs/general/next-js). A comprehensive list of SSGs available in different languages is available at [staticgen.com](http://staticgen.com).
```
